File: frontend/src/types/loss.ts
// ── Loss ─────────────────────────────────────────────────────

export interface LossEntry {
  id: number;
  machine_code: string;
  product_code: string;
  order_number: string | null;
  operator_name: string | null;
  shift: string | null;
  quantity: number;
  weight_kg: number | null;
  reason: string;
  category: string;
  material: string | null;
  is_manual: boolean;
  notes: string | null;
  timestamp: string | null;
}

export interface LossEntryCreate {
  machine_code: string;
  product_code: string;
  order_number?: string;
  operator_name?: string;
  shift?: string;
  quantity: number;
  weight_kg?: number;
  reason: string;
  category: string;
  material?: string;
  is_manual?: boolean;
  notes?: string;
}

File: synchro-mes/backend/app/schemas/loss.py
"""
Schemas: Loss entries
"""
from pydantic import BaseModel, Field
from datetime import datetime


class LossEntryCreate(BaseModel):
    machine_code: str
    product_code: str
    order_number: str | None = None
    operator_name: str | None = None
    shift: str | None = None
    quantity: int = Field(gt=0)
    weight_kg: float | None = None
    reason: str
    category: str
    material: str | None = None
    is_manual: bool = False
    notes: str | None = None


class LossEntryRead(BaseModel):
    id: int
    machine_code: str
    product_code: str
    order_number: str | None = None
    operator_name: str | None = None
    shift: str | None = None
    quantity: int
    weight_kg: float | None = None
    reason: str
    category: str
    material: str | None = None
    is_manual: bool
    notes: str | None = None
    timestamp: datetime | None = None

    model_config = {"from_attributes": True}
